Week18/App/server.js

```diff
@@ -0,0 +1,87 @@
+const express = require('express');
+const mysql = require('mysql2');
+
+const port = 3000;
+const app = express();
+
+const pool = mysql.createConnection({
+  host: 'localhost',
+  user: 'root',
+  password: 'FGSz',
+  database: 'my_app',
+  debug: false,
+});
+
+pool.connect((err) => {
+    if (err) {
+        console.error(err);
+    } else {
+        console.log("Connection established");
+    }
+});
+
+// Serve files from the 'public' folder
+app.use(express.static('public'));
+
+// Convert JSON bodies to JavaScript objects
+app.use(express.json());
+
+app.get('/', (req, res) => {
+    // Load index.html from the public folder
+    res.redirect('/index.html');
+});
+
+app.post('/api/my_app', (req, res) => {
+    const query = `
+        INSERT INTO app (dt, what, place)
+        VALUES (?, ?, ?)
+    `;
+    const params = [
+        req.body.dt,
+        req.body.what,
+        req.body.place
+    ];
+
+    pool.query(query, params, (err, result) => {
+        if (err) {
+            console.error(err);
+            res.status(500).send({ message: err.sqlMessage });
+            return;
+        }
+
+        res.status(201).send({ message: 'Success' });
+    });
+});
+
+app.get('/api/my_app', (req, res) => {
+    const query = `
+        SELECT * FROM app
+        WHERE dt BETWEEN ? AND ?
+    `;
+
+    const params = [
+        req.query.from,
```

Comment:
Érdemes lenne lekezelni azt az esetet, ha a `from` és a `to` nincs benne a query stringben.